### PR TITLE
Update Stories WordPress admin menu

### LINKED ISSUE #2750: Update Stories WordPress admin menu
## Feature Description

Ideas from the bug bash to improve the Stories Menu on the left hand side.

Stories menu (WP Left hand) could be clearer/options could be rethought. Propose having: 
1. **My Stories:** this would go to the My Stories dashboard page. 
2. **Explore templates**
3. **Add new**
4. **All Stories**
5. **Settings**

When we add other menu options (Settings, Custom Templates, etc.) we can add as appropriate. 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Update menu by adding new dumby menu items that redirect.

## Relevant Technical Choices

I was unable to cleanly re-order Add new and all stories. This would require rebuilding the menu array by editting the global, something I don't want to do / can't be done cleanly. 

To add in these menu items, I added new dumby menu items that 301 to where they need to go. 301ing is a not a great solution, as it reloads the page and does't show a current menu status. But without more work on the dashboard, I don't see how this is possible. 

Settings menu item is not displayed to users unless they have `manage_options` capability. 

Experiments was moved to the last menu item. 

 
## User-facing changes

Screenshot 2020-09-30 at 12 51 16

## Testing Instructions

1. Verify menu items are in expected order. (see screenshot above)
1. Verify menu items link to the correct pages.
1. Verify menu items are identical for users with author role and admins and work for both

---

Fixes #2750

### CODE REVIEW

**Comment by swissspidy** [comment]
> Settings menu item is not displayed to users unless they have `manage_options` capability.

Users can change their analytics opt-in on the settings page, so it should be displayed to all users with `edit_web-stories` capabilities

File: includes/Dashboard.php [hunk 1]
```diff
@@ -103,15 +103,38 @@ public function get_hook_suffix() {
...
+			'stories-dashboard-explore',
+			'__return_null',
+			1
+		);
+
+		add_submenu_page(
+			$link,
+			__( 'Settings', 'web-stories' ),
+			__( 'Settings', 'web-stories' ),
+			'manage_options',
```
Comment (spacedmonkey) [inline includes/Dashboard.php]:
```suggestion
			'edit_web-stories',
```

To confirm you want this change @swissspidy ?

File: includes/Dashboard.php [hunk 2]
```diff
@@ -103,15 +103,38 @@ public function get_hook_suffix() {
 	 * @return void
 	 */
 	public function add_menu_page() {
+		global $submenu, $menu;
```
Comment (swissspidy) [inline includes/Dashboard.php]: Seems unneeded

```suggestion
```

File: includes/Dashboard.php [hunk 3]
```diff
@@ -103,15 +103,38 @@ public function get_hook_suffix() {
...
+		$link = 'edit.php?post_type=' . Story_Post_Type::POST_TYPE_SLUG;
```
Comment (swissspidy) [inline includes/Dashboard.php]: Nit: Maybe rename to `$parent` to make it a bit more clear?

File: includes/Dashboard.php [hunk 4]
```diff
@@ -103,15 +103,38 @@ public function get_hook_suffix() {
...
 			'edit_web-stories',
 			'stories-dashboard',
 			[ $this, 'render' ],
 			0
 		);
+
+		add_submenu_page(
+			$link,
+			__( 'Explore templates', 'web-stories' ),
+			__( 'Explore templates', 'web-stories' ),
```
Comment (swissspidy) [inline includes/Dashboard.php]:
```suggestion
			__( 'Explore Templates', 'web-stories' ),
			__( 'Explore Templates', 'web-stories' ),
```

Let's use this version to match what users see on the dashboard and re-use an existing string.

(Yes I know #2750 says `Explore templates`)

File: includes/Dashboard.php (hunk 1 above)
Comment (swissspidy) [reply includes/Dashboard.php]: Yes

File: includes/Dashboard.php [hunk 5]
```diff
@@ -103,15 +103,38 @@ public function get_hook_suffix() {
 	 * @return void
 	 */
 	public function add_menu_page() {
```
Comment (swissspidy) [inline includes/Dashboard.php:111]: Tests need updating

**Comment by spacedmonkey** [comment]
@swissspidy I can't work out why e2e tests are failing here. It seems to unrelated.

**Comment by swissspidy** [comment]
@o-fernandez Is the above order of menu items acceptable? We can change the order, but it won't be that straightforward and likely error prone

**Comment by o-fernandez** [comment]
> @o-fernandez Is the above order of menu items acceptable? We can change the order, but it won't be that straightforward and likely error prone

Yes, this is fine. We can leave it like that. Thanks for checking!

**Comment by swissspidy** [comment]
> I can't work out why e2e tests are failing here. It seems to unrelated.

The `tests/e2e/specs/dashboard/documentTitle.js` test now fails because there are multiple links with the same title, and the tests probably clicks on the wrong one.

I'll check tomorrow to see if it's an error in the test or in the dashboard.

**Comment by spacedmonkey** [comment]
Can I assign this ticket to you @swissspidy as you are looking at the E2E test fail?

File: includes/Dashboard.php [hunk 6]
```diff
@@ -47,7 +47,7 @@ class Dashboard {
 	/**
 	 * Admin page hook suffix.
 	 *
-	 * @var string|false The dashboard page's hook_suffix, or false if the user does not have the capability required.
+	 * @var array|false The dashboard page's hook_suffix, or false if the user does not have the capability required.
 	 */
 	private $hook_suffix;
```
Comment (swissspidy) [inline includes/Dashboard.php]:
```suggestion
	private $hook_suffix = [];
```

File: includes/Dashboard.php [hunk 7]
```diff
@@ -47,7 +47,7 @@ class Dashboard {
 	/**
 	 * Admin page hook suffix.
 	 *
-	 * @var string|false The dashboard page's hook_suffix, or false if the user does not have the capability required.
+	 * @var array|false The dashboard page's hook_suffix, or false if the user does not have the capability required.
```
Comment (swissspidy) [inline includes/Dashboard.php]:
```suggestion
	 * @var array List of the admin pages' hook_suffixes.
```

File: includes/Dashboard.php [hunk 8]
```diff
@@ -89,10 +89,19 @@ public function init() {
...
+	 * @param string $key The current admin page key.
+	 *
+	 * @return string|false|null The dashboard page's hook_suffix, or false if the user does not have the capability required.
 	 */
-	public function get_hook_suffix() {
-		return $this->hook_suffix;
+	public function get_hook_suffix( $key ) {
+		if ( ! is_array( $this->hook_suffix ) ) {
+			return null;
+		}
```
Comment (swissspidy) [inline includes/Dashboard.php]:
```suggestion
```

File: includes/Dashboard.php [hunk 9]
```diff
@@ -103,15 +112,39 @@ public function get_hook_suffix() {
 	 * @return void
 	 */
 	public function add_menu_page() {
-		$this->hook_suffix = add_submenu_page(
-			'edit.php?post_type=' . Story_Post_Type::POST_TYPE_SLUG,
-			__( 'Dashboard', 'web-stories' ),
+		$parent = 'edit.php?post_type=' . Story_Post_Type::POST_TYPE_SLUG;
+
+		$this->hook_suffix = [];
```
Comment (swissspidy) [inline includes/Dashboard.php]:
```suggestion
```

File: includes/Dashboard.php [hunk 10]
```diff
@@ -124,8 +157,13 @@ public function add_menu_page() {
 	public function redirect_menu_page() {
```
Comment (swissspidy) [inline includes/Dashboard.php:154]: Should we exclude this method from coverage?

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
FYI, Dashboard integration tests are also failing on `main`. Known issue as per https://xwp.slack.com/archives/CCCAUAH6F/p1601955965079000

**Pull request merged**